internal/server/service/userService.go

```diff
@@ -0,0 +1,88 @@
+// Package service содержит бизнес-логику приложения GophKeeper.
+// Он включает сервисы для управления пользователями и их аутентификации.
+package service
+
+import (
+	"beliaev-aa/GophKeeper/internal/server/models"
+	"beliaev-aa/GophKeeper/internal/server/storage/repository"
+	gophKeeperErrors "beliaev-aa/GophKeeper/pkg/errors"
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"golang.org/x/crypto/bcrypt"
+)
+
+// ErrBadCredentials определяет ошибку, возникающую при неверных учетных данных для аутентификации.
+var ErrBadCredentials = errors.New("bad auth credentials")
+
+// UserService предоставляет методы для регистрации и аутентификации пользователей.
+type UserService struct {
+	userRepository *repository.UserRepository // userRepository представляет репозиторий для работы с пользователями.
+}
+
+// NewUserService создает новый экземпляр UserService с использованием заданного репозитория пользователей.
+func NewUserService(userRepository *repository.UserRepository) *UserService {
+	return &UserService{userRepository: userRepository}
+}
+
+// RegisterUser регистрирует нового пользователя в системе.
+// Принимает контекст, логин и пароль. Возвращает зарегистрированного пользователя или ошибку.
+func (s UserService) RegisterUser(ctx context.Context, login string, password string) (*models.User, error) {
```

Comment:
бы рекомендовал в общем случае использовать pointer receiver.
```suggestion
func (s *UserService) RegisterUser(ctx context.Context, login string, password string) (*models.User, error) {
```

Reply:
Исправил